MFVNeutralino/plugins/VertexTreer.cc

```diff
@@ -0,0 +1,186 @@
+#include "CommonTools/UtilAlgos/interface/TFileService.h"
+#include "DataFormats/Math/interface/deltaR.h"
+#include "DataFormats/VertexReco/interface/Vertex.h"
+#include "DataFormats/VertexReco/interface/VertexFwd.h"
+#include "FWCore/Framework/interface/EDAnalyzer.h"
+#include "FWCore/Framework/interface/Event.h"
+#include "FWCore/Framework/interface/MakerMacros.h"
+#include "FWCore/ServiceRegistry/interface/Service.h"
+#include "JMTucker/Tools/interface/ExtValue.h"
+#include "JMTucker/Tools/interface/PairwiseHistos.h"
+#include "JMTucker/Tools/interface/Utilities.h"
+#include "JMTucker/MFVNeutralinoFormats/interface/Event.h"
+#include "JMTucker/MFVNeutralinoFormats/interface/VertexAux.h"
+#include "TLorentzVector.h"
+#include "TTree.h"
+
+struct eventInfo
+{
+  int evt;
+  double weight;
+  std::vector <int> vtx_ntrack;
+  std::vector <double> vtx_dBV;
+  std::vector <double> vtx_err_dBV;
+  std::vector <double> vtx_x;
+  std::vector <double> vtx_y;
+  std::vector <double> vtx_z;
+  std::vector <double> vtx_px;
+  std::vector <double> vtx_py;
+  std::vector <double> vtx_pz;
+  std::vector <double> vtx_E;
+  std::vector <std::vector<float>> vtx_tk_dxy;
+  std::vector <std::vector<float>> vtx_tk_dxyerr;
+  std::vector <std::vector<float>> vtx_tk_dxynsigma;
+  std::vector <double> gen_vtx_x;
+  std::vector <double> gen_vtx_y;
+  std::vector <double> gen_vtx_z;
+  std::vector <double> gen_dBV;
+  std::vector <double> gen_pt;
+};
+
+class MFVVertexTreer : public edm::EDAnalyzer {
```

Comment:
Out of curiosity, what is this VertexTreer used for? Are these the trees used for the ML training, or for other studies that you've been doing? I imagine these could be useful for other studies too!

Reply:
This is used for ML, I also have some other modules that make similar trees for jets and tracks. And I think so it can be useful for other studies but most studies I did are using Histos.py.